misk-admin/src/main/kotlin/misk/web/metadata/config/ConfigMetadataAction.kt

```diff
@@ -21,58 +21,38 @@ class ConfigMetadataAction @Inject constructor(
   deployment: Deployment,
   config: Config,
   private val jvmMetadataAction: JvmMetadataAction,
-  private val mode: ConfigTabMode,
+  private val mode: ConfigTabMode
 ) : WebAction {
-  private val resources: Map<String, String?> =
-    generateConfigResources(appName, deployment, config)
+  private val resources: Map<String, String?> = generateConfigResources(appName, deployment, config)
 
   @Get("/api/config/metadata")
   @RequestContentType(MediaTypes.APPLICATION_JSON)
   @ResponseContentType(MediaTypes.APPLICATION_JSON)
   @AdminDashboardAccess
   fun getAll(): Response {
-    // TODO(mmihic): Need to figure out how to get the overrides.
-    // TODO make this config variable specific redaction with an annotation, not just password/passphrase
-    // Regex to match on password values for password redaction in output
-    val yamlFilesRegex = Regex("(?<=(password|passphrase): )([^\n]*)")
-
-    return Response(
-      resources = redact(resources, yamlFilesRegex)
-    )
+    return Response(resources = resources)
   }
 
   private fun generateConfigResources(
     appName: String,
     deployment: Deployment,
     config: Config
   ): Map<String, String?> {
+    // TODO(mmihic): Need to figure out how to get the overrides (ie. k8s /etc/secrets, database override...).
     val rawYamlFiles = MiskConfig.loadConfigYamlMap(appName, deployment, listOf())
-    val yamlFiles = linkedMapOf<String, String?>()
-    when (mode) {
-      ConfigTabMode.SAFE -> {
-        yamlFiles.put("JVM", jvmMetadataAction.getRuntime())
-      }
-      ConfigTabMode.UNSAFE_LEAK_MISK_SECRETS -> {
-        yamlFiles.put("Effective Config", MiskConfig.toYaml(config, ResourceLoader.SYSTEM))
-        rawYamlFiles.map { yamlFiles.put(it.key, it.value) }
-        yamlFiles.put("JVM", jvmMetadataAction.getRuntime())
-      }
+
+    val configFileContents = linkedMapOf<String, String?>()
+    configFileContents.put("Effective Config", MiskConfig.toRedactedYaml(config, ResourceLoader.SYSTEM))
+    if (mode == ConfigTabMode.UNSAFE_LEAK_MISK_SECRETS) {
+      rawYamlFiles.forEach { configFileContents.put(it.key, it.value) }
     }
-    return yamlFiles
+    configFileContents.put("JVM", jvmMetadataAction.getRuntime())
```

Comment:
The only use of this is to display to service owners in their admin dashboard...Since we wouldn't ever run the unsafe mode in a real production environment, I would say we should just remove any JVM metadata we don't want exposed in the admin console/endpoint.

Though, I would think our admin console ACLs should be sufficient mitigation for this concern. Anyone who is ACL'd to the admin console is an insider and could therefore determine this information out of band if they really wanted to.

Reply:
The JVM config was present in both safe/unsafe (see L53 and L58 of old file). I agree with @cconroy that the admin ACL is probably sufficient protection. I'm open to redacting any fields in JVM that you'd like to redact from engineers.

Reply:
> Anyone who is ACL'd to the admin console is an insider and could therefore determine this information out of band if they really wanted to.
We've seen at least 1 incident where admin dashboards were opened to the public web. Luckily it was for a benign service. It made me think of defense in depth here. Not a hard requirement for me